keps/349-avoid-coscheduling/README.md

```diff
@@ -0,0 +1,418 @@
+
+<!--
+Describe what tests will be added to ensure proper quality of the enhancement.
+
+After the implementation PR is merged, add the names of the tests here.
+-->
+
+### Graduation Criteria
+
+<!--
+
+Clearly define what it means for the feature to be implemented and
+considered stable.
+
+If the feature you are introducing has high complexity, consider adding graduation
+milestones with these graduation criteria:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+-->
+
+N/A
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+Delaying of workload admission until all pods are scheduled may decrease
+thoughput significantly. Especially, if there is enough resource capacity to
+which could be otherwise used to start multiple jobs at the same time.
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+#### Timeout on reaching the PodsScheduled condition
+
+We considered introduction of a timeout  to terminate any workloads which don't
+reach the `PodsScheduled` condition within the specified time since unsuspending
+of the corresponding job.
+
+If the annotation is not specified the timeout fallbacks to `1s` per pod + `30s`
+of overhead. For example, the timeout for a workload comprising 100 pods would
+fallback to 130s.
+
+**Reasons for discarding/deferring**
+
+It is also an issue currently, that admitted workloads which fail to schedule
+their pods may block the queue resources indefinetely, blocking further
+admission. Thus, it makes sense to do it as a seperate KEP.
+
+#### Delay job start instead of workload admission
+
+When a workload with `waitForPodsScheduled=true` is nominated its admission is
+blocked (rejected) until all the already admitted workloads are in the `PodsScheduled`
+condition. Instead, we could admit the workload, but delay its job start
+until the condition is satisfied.
```

Comment:
Do you mean the kueue job controller delays unsuspending the job until all other previously admitted workloads reaches the `PodsScheduled` condition? If so, the main reason for discarding this option is leaking implementation details of the feature to the job controller.

Reply:
Extended the sentence to clarify.